### PR TITLE
nep29: drop py37 support

### PR DESCRIPTION
**Pull request by bjlittle**
### Description
This PR drops support for `py37` as per the [NEP29 backwards compatibility and deprecation policy support schedule](https://numpy.org/neps/nep-0029-deprecation_policy.html#support-table).

Also see [Python endoflife](https://endoflife.date/python).

---
[Consult Iris pull request check list]( https://scitools-iris.readthedocs.io/en/latest/developers_guide/contributing_pull_request_checklist.html)

### CODE REVIEW

File: docs/src/whatsnew/latest.rst [hunk 1]
```diff
@@ -164,7 +165,8 @@ This document explains the changes made to Iris for this release
 🔥 Deprecations
 ===============
 
-#. N/A
+#. `@bjlittle`_ dropped support for ``Python 3.7``, as per the `NEP-29`_
+   deprecation policy schedule. (:pull:`4481`)
```
Comment (rcomer) [inline docs/src/whatsnew/latest.rst:169]: Deprecation or dependency? 🤔

**Comment by rcomer** [comment]
Do you need to remove the [lockfile](https://github.com/SciTools/iris/blob/main/requirements/ci/nox.lock/py37-linux-64.lock) too, or will the action tidy that up?

File: docs/src/whatsnew/latest.rst (hunk 1 above)
Comment (bjlittle) [reply docs/src/whatsnew/latest.rst]: Good call, I've moved to under the `Dependencies` section instead :+1:

**Comment by bjlittle** [comment]
> Do you need to remove the [lockfile](https://github.com/SciTools/iris/blob/main/requirements/ci/nox.lock/py37-linux-64.lock) too, or will the action tidy that up?

@rcomer The fix for the latest nox.lock and `numpy` 1.22 will be in a following PR :+1:

**Comment by rcomer** [comment]
> > Do you need to remove the [lockfile](https://github.com/SciTools/iris/blob/main/requirements/ci/nox.lock/py37-linux-64.lock) too, or will the action tidy that up?
> 
> @rcomer The fix for the latest nox.lock and `numpy` 1.22 failure will be in a following PR 👍

Yep, the numpy change affects the py38 tests and so I agree is a separate issue.  But we also have a py37 lockfile that this PR makes redundant.

**Comment by rcomer** [comment]
Another line here 👀

https://github.com/bjlittle/iris/blob/583b759a2885c8cb9977e3b57ad62c512c7d472e/.github/workflows/refresh-lockfiles.yml#L74

**Comment by bjlittle** [comment]
We'll see if the GH tidies the `py37` lock file automagically, otherwise we can nuke by PR

**Comment by pp-mo** [comment]
LGTM, but @rcomer did all the work !

**Review by pp-mo — APPROVED** [review]
LGTM, but @rcomer did all the work !

**Comment by zklaus** [comment]
Heads up, @ESMValGroup/esmvaltool-coreteam.

**Pull request merged**